runtime/service/npud/core/DBus.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONE_SERVICE_NPUD_CORE_DBUS_H__
+#define __ONE_SERVICE_NPUD_CORE_DBUS_H__
+
+#include <dbus-core.h>
+#include <gio/gio.h>
+#include <memory>
+#include <atomic>
+
+namespace npud
+{
+namespace core
+{
+
+class DBus
+{
+public:
+  DBus() noexcept;
+  ~DBus() noexcept;
+
+  DBus(const DBus &) = delete;
+  DBus &operator=(const DBus &) = delete;
+
+  bool isReady() { return _isReady.load(); }
+
+  static void on_bus_acquired(GDBusConnection *conn, const gchar *name, gpointer user_data);
+  static void on_name_acquired(GDBusConnection *conn, const gchar *name, gpointer user_data);
+  static void on_name_lost(GDBusConnection *conn, const gchar *name, gpointer user_data);
+
+  static gboolean on_handle_device_get_available_list(NpudCore *object,
+                                                      GDBusMethodInvocation *invocation);
+
+private:
+  guint _gdbus_id;
+  static std::atomic_bool _isReady;
```

Comment:
Just my curious question.
Why should this member be static?

Reply:
Because this member uses in static function. It is used in `DBus::on_bus_acquired` static function.

Reply:
My question was meant to include that static function. The function is used only in constructor of `class DBus`, so I don't understand why the function and the member should be static.

Reply:
DBus callback should be static function. So the member variable should be static too.
reference: http://www.manpagez.com/html/gio/gio-2.28.6/gio-Owning-Bus-Names.php#GBusAcquiredCallback

Reply:
I found why the callback function should be static. It's because of this pointer of DBus instance.
In other words, the static function is the type `void (*)(GDBusConnection *conn, const gchar *name, gpointer user_data)`, but `void(DBus::*)(GDBusConnection *conn, const gchar *name, gpointer user_data)` if the function is not static.

I'm sorry to bother you, but as static was added, the implementation of `DBus` was strange, so I asked. I thought It's likely that you may have considered the case where `DBus` instances are created multiple times. Anyway LGTM enough for now as is.
And it seems that `_isReady` doesn't have to be `std::atomic_bool` if there is no case where `_isReady` is set more than once.